ethash/src/compute.rs

```diff
@@ -94,8 +95,8 @@ pub struct Light {
 /// Light cache structure
 impl Light {
 	/// Create a new light cache for a given block number
-	pub fn new(block_number: u64) -> Light {
-		light_new(block_number)
+	pub fn new(cache_dir: &Path, block_number: u64) -> Light {
```

Comment:
probably `new<P: AsRef<Path>> (cache_dir: P, ...)` is more public-api friendly, along with methods down the stack